app/services/BankAccountDetailsService.scala

```diff
@@ -38,32 +38,26 @@ class BankAccountDetailsService @Inject()(val vatRegConnector: VatRegistrationCo
 
   def saveHasCompanyBankAccount(hasBankAccount: Boolean)
                                (implicit hc: HeaderCarrier, profile: CurrentProfile, ex: ExecutionContext): Future[BankAccount] = {
-    val bankAccount = if (hasBankAccount) {
-      fetchBankAccountDetails map {
-        case Some(bankAccountDetails) => bankAccountDetails.copy(isProvided = true)
-        case None => BankAccount(hasBankAccount, None, None, None)
-      }
-    } else {
-      Future.successful(BankAccount(isProvided = false, None, None, None))
+    val bankAccount = fetchBankAccountDetails map {
+      case Some(bankAccountDetails) => bankAccountDetails.copy(isProvided = hasBankAccount)
+      case None => BankAccount(hasBankAccount, None, None, None)
```

Comment:
I think this should be enhanced to properly clear down data (if it exists already) depending on aswer. I.e. if you answer true, we should set the "reason bank not provided" to None, if you answer false we should set the details (both overseas and not) to None.

That's partially what it was doing before, it was just way too aggressive in the false case and did not do it at all in the true case.

Reply:
Ah I missed the thing in NoUKBankAccountController. And the details stuff is handled by the relevant defs. Fair enough.